### ISSUE TITLE
add copyright notices to all files

### ISSUE THREAD

**Issue by MCWertGaming** [jupyter-xeus/cpp-terminal#154]
I think we should add small copyright headers to the header files. Something like:
```
/* =================
 * CPP-Terminal
 * A small library for controling modern terminals
 * (c) 2021 Ondrey Certik
 * /
```

**Comment by flagarde** [comment]
Hi @certik , I would like to solve this issue and for this would like to know your opinion about some change.

I found auto-check and auto adding copyright for source code using pre-commit, this will simplicy a lot the problem

It seems we can use this too : https://reuse.software/faq/ that means 
1) Move the MIT license inside LICENSES (It seems that the one you choose for cpp-terminal)
2) Keep LICENSE but only copy verbatim the MIT license and move the others to LICENSES/folder
For now github is not able to detect the project is MIT because of the copyright file being copy pasted in LICENSE. This avois the project to be found when searching project with MIT license 
3) use pre-commit to add the copyright

How do you feel about this move ?

**Comment by certik** [comment]
I am fine with LICENSE / LICENSES I think. I think it's fine to just use one file, but if you feel strongly you prefer two, you can.

I would not put any copyright into files. I think it's not worth the hassle. I would definitely not put names, such as mine, into files, as I think it discourages others to contribute, as they might naturally think that the person listed "owns" the file, while that is not the case.

**Comment by flagarde** [comment]
I understand your concerns, however some companies or institutions are very picky about copyright stuffs (even if the license are compatibles) and as such ask files to have copyright in it to perform automatic checks etcetc. I discovered this when some people asked me for a part of some code I wrote (and I would definitivelly think anyone would have thought it would be in public domain given the simplicity of it). That why The copyright intro is very short :
```cpp
/* cpp-terminal
* C++ library for writing multiplatform terminal applications.
*
* SPDX-FileCopyrightText: 2019-2023 Ondřej Čertík
*
* SPDX-License-Identifier: MIT
* /
```

If you don't want to have your name I think it's fine to put the organization name jupyter-xeus.
